Sources/Jaeger/Jaeger/Agent/CoreData/AgentCoreDataObjectModel.swift

```diff
@@ -0,0 +1,41 @@
+//
+//  AgentCoreDataObjectModel.swift
+//  Jaeger
+//
+//  Created by Simon-Pierre Roy on 2/1/19.
+//
+
+import CoreData
+
+/// The Core Data model for the Agent written in Code.
+enum AgentCoreDataObjectModel {
+
+    /// The Core Data model for the Agent written in Code.
+    static let model: NSManagedObjectModel = {
```

Comment:
Is it worth naming this `sharedModel` or something more singleton-y sounding, considering this is a singleton?

Reply:
yes!

Reply:
The only problem with that (in my opinion) is that `shared` typically means that the singleton instance is the same type as the containing type. eg. `URLSession.shared` is a `URLSession`.